File: swift/ReplaceElementsWithGreatestElementOnRightSide.swift
// https://leetcode.com/problems/replace-elements-with-greatest-element-on-right-side/

class Solution {
    func replaceElements(_ arr: [Int]) -> [Int] {
        var arr = arr
        var buff = -1
        for i in stride(from: arr.count - 1, through: 0, by: -1) {
            if arr[i] > buff {
                (arr[i], buff) = (buff, arr[i]) // Swap arr[i] and buff
            } else {
                arr[i] = buff
            }
        }
        return arr
    }
}

File: c/ReplaceElementsWithGreatestElementOnRightSide.c
// https://leetcode.com/problems/replace-elements-with-greatest-element-on-right-side/

int* replaceElements(int* arr, int arrSize, int* returnSize) {
    int tmp = -1, swap;
    for (int i = arrSize - 1; i >= 0; i--)
        if (tmp < arr[i]) {
            swap = arr[i];
            arr[i] = tmp;
            tmp = swap;
        } else
            arr[i] = tmp;
    *returnSize = arrSize;
    return arr;
}